### ISSUE TITLE
Add invalid handling to calcite-input-date-picker

### ISSUE THREAD

**Issue by paulcpederson** [Esri/calcite-design-system#3526]
### Description

When a user enters totally invalid input into a date picker, no feedback is given to the user or to the app developer. 

### Acceptance Criteria

- add `status` prop to calcite-input-date-picker (`status: "idle" | "invalid" | "valid"`)
- new `CalciteInputDatePickerInvalid` event added to calcite-input (emitted on blur of input with invalid date entered)

#### Which Component

`calcite-input-date-picker`

#### Example Use Case

In our application, we need a way to notify the user that the date they entered is not valid. 

@benelan feel free to assign me, happy to add this in

@jcfranco @driskull would love feedback on this api as well

**Comment by jcfranco** [comment]
Validation is something we want to support consistently across our form/value components. I'd like for the design team to look at this holistically before we start thinking about implementation. cc @HeathMeyette @macandcheese @benelan

**Comment by yelenakreyndel** [comment]
@ashetland is this item similar to the one you are working on?

**Comment by ashetland** [comment]
> @ashetland is this item similar to the one you are working on?

No, this is a separate issue and component. Not related to the input-time enhancement.

**Comment by geospatialem** [comment]
This enhancement was mentioned by the MapViewer team as well to accommodate start/end dates for the time slider.

![Mapviewer timeslider options with start date and end date validation](https://user-images.githubusercontent.com/5023024/190504232-fd79e36e-918b-4afe-9560-1fdfb7b342d2.png)

**Comment by Olga8614** [comment]
We verified this defect on ArcGIS Online 2023.r01 devext and qaext site (2/13/2023) in EN language with Edge chromium/Chrome/FF/Safari browsers.
For Gallery page, design changed, date under "Custom range..." on Gallery page can only be modified in calendar, selected date can only be showed in calendar without date textbox.
For Content/Group/Organization page: The deleted value in date is removed accordingly each time press backspace keyboard and no invalid date is appeared after deleted the date.
So this defect is fixed. Thanks!

**Comment by geospatialem** [comment]
Effort should be executed with https://github.com/Esri/calcite-design-system/issues/4276

More info on [Figma](https://www.figma.com/file/5QDPzXr21qQcPTkXR1Ouga/Form-Validation-%5BEpic-4598%5D?type=design&node-id=956-94284&mode=design&t=tPqHJcVdxmJmZ6L2-0) and below:

![image](https://github.com/Esri/calcite-design-system/assets/5023024/d696f6fc-8cd0-41f9-963b-984fdde35b2e)

**Comment by amring1** [comment]
Analysis is also tracking this fix (https://devtopia.esri.com/WebGIS/arcgis-web-analysis/issues/4166)

**Comment by SkyeSeitz** [comment]
Thanks, @geospatialem! Doesn't look like is there is any design work needed on the Online bug report, correct? I would consider this bug as separate from the validation support effort.

**Comment by geospatialem** [comment]
Opened https://github.com/Esri/calcite-design-system/issues/7588 to separate the bug from the enhancement effort to prioritize the bug effort. Will be reallocating the enhancement effort into the upcoming September milestone, and hope to tackle the bug as soon as possible. cc @banuelosj @amring1

**Comment by Olga8614** [comment]
We verified this defect on ArcGIS Online 2024.r01 devext site (12/19/2023) in ENU language with Edge Chromium/FF/Safari/Chrome browsers. Invalid characters can be entered in textbox of Start date/End date/Start time/End time/Count options on Time slider options pane. So this defect is not fixed. Thanks!

**Comment by geospatialem** [comment]
Closing in favor of the efforts completed in https://github.com/Esri/calcite-design-system/issues/8057, where the status property was added, along with an internal input-message for displaying validation messages.

Additional efforts will be completed in https://github.com/Esri/calcite-design-system/issues/8000 for setting the validity state of the component internally.

**Comment by geospatialem** [comment]
> We verified this defect on ArcGIS Online 2024.r01 devext site (12/19/2023) in ENU language with Edge Chromium/FF/Safari/Chrome browsers. Invalid characters can be entered in textbox of Start date/End date/Start time/End time/Count options on Time slider options pane. So this defect is not fixed. Thanks!

@Olga8614 While a portion of the effort has been mitigated, there are still additional requirements needed to be mitigated in full, which will be handled via #8000.
